includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php

```diff
@@ -621,6 +621,34 @@ public function trash( $post_id ) {
 		}
 	}
 
+	/**
+	 * Get data type ID for a given field.
+	 *
+	 * Possible values:
+	 *  1 = Text Field,
+	 *  2 = Text Box (textarea),
+	 *  3 = Checkbox,
+	 *  4 = Radio,
+	 *  5 = Dropdown,
+	 *  6 = Hidden field,
+	 *  7 = List Box,
+	 *  9 = Date
+	 *
+	 * @param string $field_name The field name.
+	 *
+	 * @return int Data type ID.
+	 */
+	private static function get_metadata_type( $field_name ) {
+		switch ( $field_name ) {
```

Comment:
There will be more date fields, and they'll all end with `Date`. WDYT about a regex check (`/NP_.*Date$/`) instead of a `switch`?

Reply:
I think it's a fragile assumption to put in code

Reply:
It's an assumption we can make if we set the field names. But there's also readability value in an explicit list.